src/tacacs/nss/0007-Add-support-for-TACACS-source-address.patch

```diff
@@ -0,0 +1,71 @@
+From cdd94b943a6a740070f96cb1b8af12187727efb8 Mon Sep 17 00:00:00 2001
+From: Venkatesan Mahalingam <venkatesan_mahalinga@dell.com>
+Date: Mon, 15 Jun 2020 21:13:44 -0700
+Subject: [PATCH] Add support for TACACS+ source address.
+
+Signed-off-by: Venkatesan Mahalingam <venkatesan_mahalinga@dell.com>
+---
+ nss_tacplus.c | 19 ++++++++++++++++++-
+ 1 file changed, 18 insertions(+), 1 deletion(-)
+
+diff --git a/nss_tacplus.c b/nss_tacplus.c
+index 7feeda6..d7c68d3 100644
+--- a/nss_tacplus.c
++++ b/nss_tacplus.c
+@@ -73,6 +73,7 @@ typedef struct {
+ static tacplus_server_t tac_srv[TAC_PLUS_MAXSERVERS];
+ static int tac_srv_no;
+ static useradd_info_t useradd_grp_list[MAX_TACACS_USER_PRIV + 1];
++static struct addrinfo *source_addr;
+ 
+ static char *tac_service = "shell";
+ static char *tac_protocol = "ssh";
+@@ -247,6 +248,8 @@ static int parse_config(const char *file)
+         return NSS_STATUS_UNAVAIL;
+     }
+ 
++    if(source_addr)
++        freeaddrinfo(source_addr);
```

Comment:
Being static member, will there be a stale address usage problem, unless you explicitly set it to NULL upon free?

Reply:
Set source_addr to NULL after free.